indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/X509SignatureAlgorithm.kt

```diff
@@ -91,35 +94,85 @@ enum class X509SignatureAlgorithm(
             ES256, PS256, RS256 -> Digest.SHA256
             ES384, PS384, RS384 -> Digest.SHA384
             ES512, PS512, RS512 -> Digest.SHA512
+            else -> throw IllegalArgumentException("Unsupported hash algorithm.")
         }
 
     override val algorithm: SignatureAlgorithm
         get() = when (this) {
-            ES256, ES384, ES512 -> SignatureAlgorithm.ECDSA(this.digest, null)
-            PS256, PS384, PS512 -> SignatureAlgorithm.RSA(this.digest, RSAPadding.PSS)
-            RS1, RS256, RS384, RS512 -> SignatureAlgorithm.RSA(this.digest, RSAPadding.PKCS1)
+            is EC -> SignatureAlgorithm.ECDSA(digest, null)
+            is RSA -> SignatureAlgorithm.RSA(digest, if (pssBits != null) RSAPadding.PSS else RSAPadding.PKCS1)
+            else -> throw IllegalArgumentException("Unsupported signature algorithm.")
```

Comment:
maybe leave a comment here so that we don't miss updating this when we make our core data classes equally extensible?